R/update-input.R

```diff
@@ -452,16 +452,18 @@ updateSliderInput <- function(session, inputId, label = NULL, value = NULL,
 
 
 updateInputOptions <- function(session, inputId, label = NULL, choices = NULL,
-                               selected = NULL, inline = FALSE,
-                               type = 'checkbox') {
-  if (!is.null(choices))
-    choices <- choicesWithNames(choices)
+                               selected = NULL, inline = FALSE, type = NULL,
+                               choiceNames = NULL, choiceValues = NULL) {
+  args <- normalizeChoicesArgs(choices, choiceNames, choiceValues)
+
   if (!is.null(selected))
-    selected <- validateSelected(selected, choices, session$ns(inputId))
+    selected <- normalizeSelected(selected, session$ns(inputId),
+      args$choiceNames, args$choiceValues)
 
-  options <- if (!is.null(choices)) {
+  options <- if (!is.null(args$choiceValues)) {
```

Comment:
This should be `if (!is.null(choices) || !is.null(choiceNames) || !is.null(choiceValues))`.